### PR TITLE
[release-1.5] [backport] Move all E2E test templates to use external CCM and CSI

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind backport
/kind support

**What this PR does / why we need it**:
Backport of #4005 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2941

**Special notes for your reviewer**:
Scope of this PR is to fix regular CAPA managed E2E tests. EKS tests have also been failing from long time, but it is out of scope of this PR.

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [X] adds or updates e2e tests

**Release note**:

```release-note
Move all E2E test templates to use external CCM and CSI
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
Thanks @Skarlso for pointing out the root cause. 🎉

File: test/e2e/data/infrastructure-aws/kustomize_sources/efs-support/machine-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    spec:
+      clusterName: "${CLUSTER_NAME}"
+      version: "${KUBERNETES_VERSION}"
+      bootstrap:
+        configRef:
+          name: "${CLUSTER_NAME}-md-0"
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfigTemplate
+      infrastructureRef:
+<<<<<<<< HEAD:test/e2e/data/infrastructure-aws/kustomize_sources/gpu/machine-deployment.yaml
```
Comment (Skarlso) [inline test/e2e/data/infrastructure-aws/kustomize_sources/efs-support/machine-deployment.yaml:21]: @Ankitasw Ops. :D

Comment (Ankitasw) [reply test/e2e/data/infrastructure-aws/kustomize_sources/efs-support/machine-deployment.yaml:21]: Thanks 😓

**Comment by Ankitasw** [comment]
There might be more errors due to cherry pick, which I could resolve only after the runs

**Comment by Ankitasw** [comment]
cc @Skarlso @richardcase 
Can someone please take a look?

**Comment by Skarlso** [comment]
/lgtm
as far as I can tell. :D

**Comment by Ankitasw** [comment]
Tests are failing in current run 😢 so I have to check again

**Comment by Ankitasw** [comment]
E2E tests are passing now 🎉

File: test/e2e/data/infrastructure-aws/kustomize_sources/nested-multitenancy-clusterclass/role.yaml [hunk 2]
```diff
@@ -1,25 +0,0 @@
----
```
Comment (richardcase) [inline test/e2e/data/infrastructure-aws/kustomize_sources/nested-multitenancy-clusterclass/role.yaml:1]: How come this was deleted?It looks like its still being reference from the kustomization.yaml

Comment (Ankitasw) [reply test/e2e/data/infrastructure-aws/kustomize_sources/nested-multitenancy-clusterclass/role.yaml:1]: Does that make a difference? I think yaml is still getting applied.

Comment (Ankitasw) [reply test/e2e/data/infrastructure-aws/kustomize_sources/nested-multitenancy-clusterclass/role.yaml:1]: I think this was unintentional, but I am surprised how its working in E2E

**Comment by Ankitasw** [comment]
PR is ready for review.

**Comment by Ankitasw** [comment]
cc @Skarlso @richardcase for final review

**Pull request merged**